plugins/chat/assets/javascripts/discourse/components/chat-thread-title-prompt.js

```diff
@@ -0,0 +1,125 @@
+import { setOwner } from "@ember/application";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { NotificationLevels } from "discourse/lib/notification-levels";
+import I18n from "I18n";
+import ThreadSettingsModal from "discourse/plugins/chat/discourse/components/chat/modal/thread-settings";
+import { THREAD_TITLE_PROMPT_THRESHOLD } from "discourse/plugins/chat/discourse/lib/chat-constants";
+import UserChatThreadMembership from "discourse/plugins/chat/discourse/models/user-chat-thread-membership";
+
+export default class ShowThreadTitlePrompt {
+  @service chatApi;
+  @service modal;
+  @service toasts;
+  @service currentUser;
+  @service site;
+
+  toastText = {
+    title: I18n.t("chat.thread_title_toast.title"),
+    message: I18n.t("chat.thread_title_toast.message"),
+    dismissLabel: I18n.t("chat.thread_title_toast.dismiss_action"),
+    primaryLabel: I18n.t("chat.thread_title_toast.primary_action"),
+  };
+
+  constructor(owner, thread) {
+    setOwner(this, owner);
+    this.thread = thread;
+    this.createThreadMembership();
+  }
```

Comment:
Once you have added the component to chat-thread.gjs as I mentioned in another comment, here you will do this logic:

if (this.shouldShowToast) {
  this.show();
  this.updateThreadTitlePrompt();
}